File: Assets/Scripts/Effects/FaceToCameraObjPrepareData.cs
using System;
using UnityEngine;
using System.Collections.Generic;

public class FaceToCameraObjPrepareData : MonoBehaviour
{
    [SerializeField]
    private FaceToCameraObj[] _faceToCameraObjs;
    private object _skinnedMeshRenderers;

    public FaceToCameraObj[] FaceToCameraObjs
    {
        get
        {
            return _faceToCameraObjs;
        }
        set
        {
            _faceToCameraObjs = value;
        }
    }

    public void CustomActive(bool active)
    {
        if (_faceToCameraObjs != null)
        {
            foreach(var com in _faceToCameraObjs)
            {
                com.enabled = active;
            }
        }
    }
}

File: Assets/Scripts/Lua/UnityLuaAPI/FaceToCameraObjPrepareData.lua
---@class FaceToCameraObjPrepareData : UnityEngine.MonoBehaviour
---@field public FaceToCameraObjs FaceToCameraObj[]

---@type FaceToCameraObjPrepareData
FaceToCameraObjPrepareData = { }
---@return FaceToCameraObjPrepareData
function FaceToCameraObjPrepareData.New() end
---@param active boolean
function FaceToCameraObjPrepareData:CustomActive(active) end
return FaceToCameraObjPrepareData
